shm/Island.hpp

```diff
@@ -1,27 +1,35 @@
 #pragma once
 #include <cstdint>
+#include <iostream>
 #include <string>
-
-using coordinateType = std::size_t;
-
 class Island {
 public:
+    using coordinateType = std::size_t;
+
     Island(coordinateType positionX, coordinateType positionY);
 
     class Coordinates {
     public:
-        Coordinates(coordinateType positionsX, coordinateType positionY);
-        std::string toString();
+        constexpr Coordinates(Island::coordinateType positionX, Island::coordinateType positionY)
+            : positionX_{positionX}, positionY_{positionY} {}
 
-        bool operator==(const Coordinates& other) const;
+        constexpr bool operator==(const Coordinates& other) const {
+            return positionX_ == other.positionX_ and positionY_ == other.positionY_;
+        }
+
+        friend std::ostream& operator<<(std::ostream& out, const Coordinates& coords);
 
     private:
-        const coordinateType positionX_;
-        const coordinateType positionY_;
+        const Island::coordinateType positionX_ = 0;
+        const Island::coordinateType positionY_ = 0;
```

Comment:
You don't have default C'tor. So you always will initialize these values. That's why you don't need to assign default values here.